src/App.jsx

```diff
@@ -14,45 +15,87 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_BY_ALPHABET = 'alphabet';
+const SORT_BY_LENGTH = 'length';
+
+const sorted = (goods, sortBy, isReverse) => {
+  let prepearedGoods = [...goods];
+
+  if (sortBy) {
+    prepearedGoods.sort((good1, good2) => {
+      switch (sortBy) {
+        case SORT_BY_ALPHABET:
+          return good1.localeCompare(good2);
+        case SORT_BY_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReverse) {
+    prepearedGoods = prepearedGoods.reverse();
+  }
+
+  return prepearedGoods;
+};
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [goodsReverse, setGoodsReverse] = useState(false);
+  const visibleGoods = sorted(goodsFromServer, sortField, goodsReverse);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const handleReset = () => {
+    setSortField('');
+    setGoodsReverse(false);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => setSortField(SORT_BY_ALPHABET)}
+          type="button"
+          className={`button is-info ${sortField !== SORT_BY_ALPHABET && 'is-light'}`}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          onClick={() => setSortField(SORT_BY_LENGTH)}
+          type="button"
+          className={`button is-success ${sortField !== SORT_BY_LENGTH && 'is-light'}`}
+        >
+          Sort by length
+        </button>
+
+        <button
+          onClick={() => setGoodsReverse(!goodsReverse)}
+          type="button"
+          className={`button is-warning ${!goodsReverse && 'is-light'}`}
+        >
+          Reverse
+        </button>
+
+        {(sortField || goodsReverse) && (
+          <button
+            onClick={() => handleReset()}
+            type="button"
+            className="button is-danger is-light"
+          >
+            Reset
+          </button>
+        )}
+
+      </div>
+
+      <ul>
+        {visibleGoods.map(good => (
+          <li data-cy="Good" key={good}>{good}</li>
+        ))}
```

Comment:
better to add index to good in key because may be that in your arr good with same text
```suggestion
        {visibleGoods.map((good, index) => (
          <li data-cy="Good" key={good + index}>{good}</li>
        ))}
```

Reply:
![js](https://github.com/mate-academy/react_list-of-goods-js/assets/104643607/3e3b098f-2958-4cd9-8ba4-b48ebb111fa8)

Reply:
![jsx](https://github.com/mate-academy/react_list-of-goods-js/assets/104643607/c6c5ef18-22b8-4d19-8523-5c140b6974f2)

Reply:
touchér